### PR TITLE
Binary search

### PR DESCRIPTION
**Pull request by sbshah97**
This pull request is for issue number #19 

### Brief description about contribution
* Binary Search Algorithm added using iterative version. 
* Vectors used instead of the conventional arrays.

### CODE REVIEW

**Comment by sbshah97** [comment]
@VamsiSangam kindly take a look!

**Comment by sbshah97** [comment]
Kindly merge all the commits!
Sorry for the mess! First contribution that's why! :see_no_evil:

File: Searching Algorithms/Binary Search Algorithm/C++/binary-search.cpp [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+Code Contributor - <salman-bhai>
+
+===== ===== ===== */
+//Includes all header files required for C++ functions
+#include <bits/stdc++.h>
+using namespace std;
+
+vector<int> v;
+
+//Main Binary Search logic
```
Comment (ranesr) [inline Searching Algorithms/Binary Search Algorithm/C++/binary-search.cpp]: Can you please update all comments like?
```
// Main Binary Search logic
```

File: Searching Algorithms/Binary Search Algorithm/C++/binary-search.cpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+int binary(int key) {
+	int first = 0;
+	int last = v.size()-1;
+
+	while(first <= last) {
+		int mid = ((first+last)/2);
```
Comment (VamsiSangam) [inline Searching Algorithms/Binary Search Algorithm/C++/binary-search.cpp]: Spaces on both sides of each binary operator please.

File: Searching Algorithms/Binary Search Algorithm/C++/binary-search.cpp [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+		if(v[mid] == key) {
+			return mid;
+		}
+		else if(v[mid] > key) {
+			last = mid - 1;
+		}
+		else
```
Comment (VamsiSangam) [inline Searching Algorithms/Binary Search Algorithm/C++/binary-search.cpp]: Do add braces here for better readability.

File: Searching Algorithms/Binary Search Algorithm/C++/binary-search.cpp [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+	int key;
+	cin >> key;
+
+	int x = binary(key);
+
+	//In case -1 is found, means key isn't present
+	if(x == -1)
+		printf("Key not found!\n");
+	//If -1 isn't returned, key is found!
+	else
```
Comment (VamsiSangam) [inline Searching Algorithms/Binary Search Algorithm/C++/binary-search.cpp]: Add curly braces here too.

File: Searching Algorithms/Binary Search Algorithm/C++/binary-search.cpp [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+		printf("Key found at position %d\n",x+1);
```
Comment (VamsiSangam) [inline Searching Algorithms/Binary Search Algorithm/C++/binary-search.cpp]: Spaces for binary operator.

**Review by VamsiSangam — COMMENTED** [review]
Code runs fine! Requested minor changes.

**Review by ranesr — CHANGES_REQUESTED** [review]

**Comment by sbshah97** [comment]
Changes made as per suggestions by @ranesr

**Review by VamsiSangam — COMMENTED** [review]
Actually @ranesr meant to add a space before starting a comment. So, your comments should look like this -

```
// Main binary search logic
```

and not like this -

```
//Main binary search logic
```

You can find more info about the coding conventions in [CONVENTIONS.md](https://github.com/VamsiSangam/theoryofprogramming/blob/master/CONVENTIONS.md) :smile:

**Comment by sbshah97** [comment]
How about now :sweat_smile:

**Review by VamsiSangam — APPROVED** [review]
LGTM :+1:

**Comment by VamsiSangam** [comment]
We mind a lot about the readability, so don't mind the reminders about the conventions :stuck_out_tongue_closed_eyes:

**Comment by VamsiSangam** [comment]
Thanks for your contribution, you should be able to see your code live on the [corresponding post](http://theoryofprogramming.com/2016/11/04/binary-search-algorithm/), in a few minutes.

**Comment by VamsiSangam** [comment]
I must have overlooked it. We really want the vector to be a local variable, being passed to binary search method as function parameters. I have opened an issue for that. Please work on #48 when free. :smile:

**Pull request merged**